tools/universe/s3_uploader.py

```diff
@@ -36,6 +38,55 @@ def upload(self, filepath, content_type=None):
         cmdlist.append('{} {} 1>&2'.format(filepath, dest_url))
         cmd = ' '.join(cmdlist)
         log.info(cmd)
+
+        # try once:
         ret = os.system(cmd)
-        if not ret == 0:
+        if ret != 0:
+            if self._renew_credentials():
+                # renew succeeded, try again:
+                ret = os.system(cmd)
+        if ret != 0:
+            # failed once, then renewal failed or retry failed
             raise Exception('Failed to upload {} to {}'.format(filepath, dest_url))
+
+    def _renew_credentials(self):
+        if self._reauth_attempted:
+            # reauth was already tried once during this session. any failure must be from something else.
+            return False
+        self._reauth_attempted = True
+
+        # check that maws is installed before trying anything else
+        if subprocess.run('maws -h'.split(), stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL).returncode != 0:
```

Comment:
Any reason why we don't use `which`?

Reply:
Good point, I'll fix it in a follow-up PR